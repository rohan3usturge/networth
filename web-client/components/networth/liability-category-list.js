import { LiablityCategory } from "./liability-category";
import { NUMBER_UTILS } from "./../utils";
import { useRouter } from "next/router";
import { withTranslation } from "react-i18next";

const Wrapper = ({ categories, onChange, disabled, currency, total, t }) => {
  const handleChange = (changedCategory) => {
    const changed = categories.map((category) => {
      if (category.meta.id === changedCategory.meta.id) {
        return changedCategory;
      }

      return category;
    });
    if (onChange) {
      onChange(changed);
    }
  };

  return (
    <>
      <div className="card mb-3">
        <div className="card-header bg-danger text-white">
          <h4 className="card-title">
            {NUMBER_UTILS.convertNumToMoney(
              total,
              currency,
              useRouter().locale
            )}
          </h4>
          <span className="card-text">
            <i className="fas fa-file-invoice mr-2"></i>
            {t("liabilities")}
          </span>
        </div>
        <div className="card-body">
          {categories.map((category) => {
            return (
              <LiablityCategory
                key={category.meta.id}
                currency={currency}
                category={category}
                onChange={handleChange}
                disabled={disabled}
              />
            );
          })}
        </div>
      </div>
    </>
  );
};

const LiabilityCategoryList = withTranslation()(Wrapper);

export { LiabilityCategoryList, Wrapper };
