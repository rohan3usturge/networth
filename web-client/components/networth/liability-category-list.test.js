import { shallow } from "enzyme";
import { Wrapper } from "./liability-category-list";

jest.mock("next/router", () => {
  return {
    useRouter: () => ({ locale: "en-ca" }),
  };
});

describe("LiabilityCategoryList", () => {
  it("should render properly and match snapshot", () => {
    const translations = {
      currency: "Currency",
    };

    const wrapper = shallow(
      <Wrapper
        categories={[]}
        currency="CAD"
        disabled={true}
        onChange={() => {}}
        total="2323.0232"
        key="2323"
        t={(key) => translations["currency"]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
