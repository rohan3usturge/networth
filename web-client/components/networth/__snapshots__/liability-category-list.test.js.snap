// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiabilityCategoryList should render properly and match snapshot 1`] = `
<Fragment>
  <div
    className="card mb-3"
  >
    <div
      className="card-header bg-danger text-white"
    >
      <h4
        className="card-title"
      >
        CA$2,323.02
      </h4>
      <span
        className="card-text"
      >
        <i
          className="fas fa-file-invoice mr-2"
        />
        Currency
      </span>
    </div>
    <div
      className="card-body"
    />
  </div>
</Fragment>
`;
